docs/adr/0008_signing_verification.md

```diff
@@ -0,0 +1,358 @@
+
+### Normalization
+
+- Canonicalization for component-version digest and signature payload is **JCS (RFC 8785)** via the existing `jcs` package.
+- No CLI flag to select a different normalization.
+
+---
+
+## Behavior
+
+### Digest calculation (`ocm add digests cv`)
+
+- Computes and **embeds** digests:
+  - **Resources**: content digests.
+  - **Component references**: digest of the referenced component descriptor.
+  - **Component-version**: digest over the canonicalized descriptor.
+- May download artifacts and referenced descriptors as needed.
+- Writes the updated descriptor back to `<ref>`.
+
+### Signing (`ocm sign cv`)
+
+- Loads the descriptor from `<ref>`.
+- Canonicalizes the descriptor using JCS.
+- Computes the component-version digest from canonical bytes.
+- If `--pin` is provided, compares and fails on mismatch.
+- Performs direct signing using `--cert` (or a default signer if available).
+- If `--cosign` is set:
+  - Obtain an OIDC ID token in this order:
+    1. Use `--cosign-identity-token` if provided (token string or `@path` to file).
+    2. Else use `SIGSTORE_ID_TOKEN` environment variable if present.
+    3. Else, if interactive TTY is available, run an **interactive loopback** browser flow to retrieve the token.
+    4. Else, run the **device flow** to retrieve the token.
+  - Produce a Cosign signature over the **same canonical bytes**.
+  - When `--cosign-upload` is present and `<ref>` maps to an OCI subject (`<name>@<digest>`), attach the Cosign signature to the registry using Cosign conventions.
+  - Record the **Cosign bundle** (signature, certs, and optional Rekor inclusion) in the envelope for offline verification.
+- Append the signature envelope under `.signatures[]`.
+- Do not compute or embed any digests during signing.
+
+---
+
+## CLI Examples
+
+```bash
+# Digests
+ocm add digests cv ghcr.io/org/app:1.2.3 --recurse --force
+
+# Direct sign
+ocm sign cv ghcr.io/org/app:1.2.3 \
+  --sig release@2025-08-27 \
+  --cert ~/.keys/release.p12 \
+  --password-prompt
+
+# Direct sign + embed Cosign bundle (no upload)
+ocm sign cv ghcr.io/org/app:1.2.3 \
+  --sig release@2025-08-27 \
+  --cosign
+
+# Direct sign + Cosign upload to OCI and Rekor (keyless OIDC via browser/device)
+ocm sign cv ghcr.io/org/app:1.2.3 \
+  --sig release@2025-08-27 \
+  --cosign \
+  --cosign-upload \
+  --cosign-annotation git.sha=deadbeef
+
+# Cosign with pre-fetched ID token (no browser)
+ocm sign cv ghcr.io/org/app:1.2.3 \
+  --sig rel@2025 \
+  --cosign \
+  --cosign-identity-token @/run/secure/idtoken \
+  --cosign-skip-tlog
+```
+
+---
+
+## Code Design
+
+### Normalization
+
+```go
+// Using existing package jcs (RFC 8785 canonical JSON)
+canon, err := jcs.Normalise(componentDescriptor, nil)
+if err != nil { /* handle */ }
+```
+
+### Component-version digest
+
+```go
+sum := sha256.Sum256(canon)
+componentDigest := "sha256:" + hex.EncodeToString(sum[:])
+```
+
+### Signature envelope
```

Comment:
this is not how digests are defined in OCM right now, see https://github.com/open-component-model/ocm-spec/blob/main/doc/02-processing/03-signing-process.md

TBH I dont think we should break the existing OCM signature Specification, so you would need to map this back